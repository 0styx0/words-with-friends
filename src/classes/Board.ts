import TileInfo from './TileInfo';

export default class Board extends Map {

    constructor(board?: Board) {
        super();

        // copies board
        if (board) {
            board.forEach((value, key) => super.set(key, value));
        }
    }

    set(coordinates: number[], value: TileInfo) {
        return super.set(`${coordinates[0]}, ${coordinates[1]}`, value);
    }

    get(coordinates: number[]): TileInfo | undefined {

        return coordinates ? super.get(`${coordinates[0]}, ${coordinates[1]}`) : new TileInfo();
    }
}