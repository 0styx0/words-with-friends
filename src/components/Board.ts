
class Board {

    dimensions = 15; // square board, so 15x15
}