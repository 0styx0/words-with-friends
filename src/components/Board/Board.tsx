import * as React from 'react';
import './index.css';

import BoardTileHolder from '../TileHolder/TileHolder';

export default function Board() {

    let board: typeof BoardTileHolder[] = [];

    for (let i = 0; i < +process.env.REACT_APP_BOARD_DIMENSIONS!; i++) {

        for (let j = 0; j < +process.env.REACT_APP_BOARD_DIMENSIONS!; j++) {

            board.push(
                <BoardTileHolder
                    coordinates={[j, i]}
                    key={`${j}, ${i}`}
                    {...[] as any}
                /> as {} as typeof BoardTileHolder
            );
        }
    }

    return (
        <div className="wrapper">
            {board}
        </div>
    );
}
