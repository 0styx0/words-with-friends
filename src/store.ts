import reducers from './reducers';
import { createStore, compose } from 'redux';
import Player from './classes/Player';
import Tilebag from './classes/Tilebag';
import Board from './classes/Board';

const enhancers = compose('devToolsExtension' in window ?
    (window as {} as { devToolsExtension: Function }).devToolsExtension!() :
    (f: {}) => f);

export const defaultState = {
    turn: 0,
    Players: [new Player(true, 0), new Player(false, 1)], // turns switched around on INIT_PLAYERS
    board: new Board(),
    Tilebag: new Tilebag()
};

const store = createStore(reducers, defaultState, enhancers);

export default store;
