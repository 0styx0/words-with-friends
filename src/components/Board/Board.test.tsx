import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Board from './Board';
import { Provider } from 'react-redux';
import store from '../../store';
import mockMath from '../../test/mocks/Math';
mockMath();

describe('Board', () => {

    it('renders correctly', () => {

        store.dispatch({
            type: 'INIT_BOARD'
        });

        const tree = renderer.create(
            <Provider store={store}>
                <Board />
            </Provider>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});