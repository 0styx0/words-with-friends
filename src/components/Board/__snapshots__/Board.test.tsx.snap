// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board renders correctly 1`] = `
<div
  className="wrapper"
>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
  <div
    className="tileHolder"
    onDragOver={[Function]}
    onDrop={[Function]}
  >
    <span>
      
    </span>
  </div>
</div>
`;
